src/darwin/Framework/CHIP/MTRBaseDevice.mm

```diff
@@ -1384,16 +1384,20 @@ - (instancetype)initWithPath:(const ConcreteEventPath &)path
 {
     __block NSArray * attributeReports = mAttributeReports;
     mAttributeReports = nil;
+    __block auto attributeCallback = mAttributeReportCallback;
+
     __block NSArray * eventReports = mEventReports;
     mEventReports = nil;
-    if (mAttributeReportCallback && attributeReports.count) {
+    __block auto eventCallback = mEventReportCallback;
+
+    if (attributeCallback != nil && attributeReports.count) {
         dispatch_async(mQueue, ^{
-            mAttributeReportCallback(attributeReports);
+            attributeCallback(attributeReports);
```

Comment:
I think this should check for attributeCallback != nil before calling it in here in the dispatch block.

Reply:
How would it become nil, if we checked it for non-nil before queueing the block, and it's a block variable?

Reply:
Can't it not become nil if `ReportError` got called after your block got queued for execution, but before your block gets dispatched?

Reply:
How?  It's a pointer that was on the stack and then was copied into the block.  `mAttributeReportCallback` can become null, which is precisely the thing we are guarding against by not using it inside the block.  But I don't see how `attributeCallback` can.

Or am I just totally mis-understanding how all this works?